File: BACKEND/ApiRestFull/ApiGestionAgua/ApiGestionAgua/Modelos/Cliente.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ApiGestionAgua.Modelos
{
    public class Cliente
    {

        [Key]
        public int IdCliente { get; set; }

        [Required]
        public string Calle { get; set; }

        [Required]
        public string Piso { get; set; }

        [Required]
        public string Depto { get; set; }

        [Required]
        public string Telefono { get; set; }


        public int IdUsuario { get; set; }

        [ForeignKey("IdUsuario")]
        public Usuario Usuario { get; set; }

        
        public int IdBarrio { get; set; }

        [ForeignKey("IdBarrio")]
        public Barrio barrio { get; set; }

    }
}

File: FRONTEND/pryGestionDeAgua_Cliente/src/app/Interfaces/cliente.ts
export interface Cliente {
    idCliente: number;
    calle:     string;
    piso:      string;
    depto:     string;
    telefono:  string;
    idUsuario: number;
    usuario:   Usuario;
    idBarrio:  number;
    barrio:    null;
}

export interface Usuario {
    idUsuario:       number;
    dni:             string;
    nombre:          string;
    apellido:        string;
    sexo:            string;
    mail:            string;
    password:        string;
    fechaNacimiento: Date;
    fechaAlta:       Date;
    idRol:           number;
    rol:             null;
}
